src/Uno.UI/Media/NativeRenderTransformAdapter.iOSmacOS.cs

```diff
@@ -11,25 +13,66 @@ namespace Uno.UI.Media
 {
 	partial class NativeRenderTransformAdapter
 	{
+		private CATransform3D _transform = CATransform3D.Identity;
+		private bool _wasAnimating = false;
+
 		partial void Initialized()
 		{
 			// On íOS and MacOS Transform are applied by default on the center on the view
 			// so make sure to reset it when the transform is attached to the view
-			InitializeOrigin();
+			InitializeOrigin();
+
+#if __MACOS__
+			// On MAC OS, if we set the Transform before the NSView has been rendered at least once,
+			// it will be definitively ignored (even if updated).
+			// So here we hide the control (using Layer.Opacity to avoid not alter the Element itself),
+			// and wait for the control to be loaded to set the transform and restore the visibility.
+			if (Owner is FrameworkElement element && !element.IsLoaded)
+			{
+				_isDeferring = true;
+				element.Loaded += DeferredInitialize;
+				element.Layer.Opacity = 0;
 
-			// Apply the transform as soon as its been declared
+				return;
+			}
+#endif
+
+			// Apply the transform as soon as its been declared
 			Update();
-		}
+		}
+
+#if __MACOS__
+		private bool _isDeferring;
 
-		private CATransform3D _transform = CATransform3D.Identity;
-		private bool _wasAnimating = false;
+		private static void DeferredInitialize(object sender, RoutedEventArgs e)
+		{
+			var fwElt = (FrameworkElement)sender;
+			var adapter = fwElt._renderTransform;
```

Comment:
Could this be null ?

Reply:
Not here: the `_renderTransform` it actually itself